src/scenes/titlescreen.ts

```diff
@@ -1,37 +1,45 @@
-import { Nullable, UUID } from "definitions/utils";
+import { float, Nullable } from "definitions/utils";
 import SessionManager from "managers/SessionManager";
 import Phaser from "phaser";
 import { interactify } from "../utilities/utils";
 import { ScreenHeight, ScreenWidth } from "../core/main";
+import Image = Phaser.GameObjects.Image;
 
 export class TitleScreen extends Phaser.Scene {
-    private title: Nullable<Phaser.GameObjects.Text>;
     private button: Nullable<Phaser.GameObjects.Image>;
 
     public constructor() {
         super("TitleScreen");
-        this.title = null;
         this.button = null;
     }
 
     public preload(): void {
         this.load.image("createLobby", "assets/create.png");
+        this.load.image("titlescreen", "assets/sabo.png");
     }
 
     public create(): void {
-        this.title = this.add
-            .text(ScreenWidth / 2, ScreenHeight / 4, "Saboteur", {
-                font: "Arial Black",
-                fontSize: 38,
-                color: "#ffffff",
-                stroke: "#000000",
-                strokeThickness: 8,
-                align: "center",
-            })
-            .setOrigin(0.5);
-        this.button = this.add.image(
+        this.add
+            .rectangle(0, 0, ScreenWidth, ScreenHeight, 0x000000)
+            .setOrigin(0);
+
+        const backgroundImage: Image = this.add.image(
             ScreenWidth / 2,
-            ScreenHeight / 1.5,
+            ScreenHeight / 2,
+            "titlescreen",
+        );
+
+        const maxWidthScale: float = ScreenWidth / backgroundImage.width;
+        const maxHeightScale: float = ScreenHeight / backgroundImage.height;
+        const scale: float = Math.min(maxWidthScale, maxHeightScale);
+        backgroundImage.setScale(scale);
```

Comment:
I love the dynamic solution here. May i suggest the following:
- Make the image a little bigger by adding a little more "uninteresting" rock-formations around it.
- Change the math.min to .max and the image will be full screen adaptable without weird stretching.

Reply:
Do you mean by making the image bigger, drawing more rock-formations?

Reply:
so like in your drawing software make the canvas little bigger and put your current image in the center then draw/expand/continue with more rock around it, thus creating a buffer zone which would be oke if it would be off screen. Then export it and put it into the scene and change to Math.max and the background will fill the entire screen.